txp/Pos/Txp/Toil/Logic.hs

```diff
@@ -115,13 +119,47 @@ normalizeToil txs = mapM_ normalize ordered
 
 verifyToilEnv
     :: (MonadToilEnv m, MonadError ToilVerFailure m)
-    => TxAux -> m ()
-verifyToilEnv txAux = do
-    limit <- teMaxTxSize <$> getToilEnv
-    let txSize = biSize txAux
+    => TxAux -> TxFee -> m ()
+verifyToilEnv txAux txFee = do
+    toilEnv <- getToilEnv
+    let
+        limit = teMaxTxSize toilEnv
+        mtxFeePolicy = teTxFeePolicy toilEnv
+        txSize = biSize txAux
+    case mtxFeePolicy of
+        Nothing ->
```

Comment:
Actually it should never happen (because policy is always `Just` in genesis `BlockVersionData`), but it doesn't seem easy to enforce it on type-level without something cumbersome.

It would be good to log error in such case, but not necessary (probably it will involve propagating `WithLogger` constraint into too many places).